lib/libzfs/os/linux/libzfs_mount_os.c

```diff
@@ -361,7 +361,7 @@ do_unmount(const char *mntpt, int flags)
 }
 
 int
-zfs_can_user_mount(void)
+zfs_mount_delegation_check(void)
 {
-	return (geteuid() == 0);
+	return (geteuid() != 0);
```

Comment:
Good find with the existing `zfs_can_user_mount` function.  Let's just update the new `zfs_mount_delegation_check()` to return `EACCES` on failure instead of just non-zero.